### ISSUE TITLE
Determine how screen readers should interact with disabled controls

### ISSUE THREAD

**Issue by taysea** [grommet/grommet#6089]
### Expected Behavior

Consistent screen reader / keyboard treatment of disabled items across components.

### Actual Behavior

Currently we have differing behaviors around how disabled controls are handled for screen reader, we should investigate the best practice approach and apply to interactive components as necessary (DataTable, List, Select, Menu, CheckboxGroup, etc.)

DataTable lets you move with keyboard to the disabled control so the screen reader can tell you that it's disabled (or why it's disabled).

Other components like Select skips over the disabled items when using down arrow.

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by halocline** [comment]
Findings from List:

As example, consider presenting a product on a product detail page: 
- The product may have attributes such as color and size. 
- The color and sizes options are presented in a list so that the user knows what options are available. 
- However, at certain points in time, a particular color or size may be out of stock and the interface will want to display that item as disabled, indicating that that item is normally carried, but currently out of stock.

Given the above situation and those that are similar:
- Disabled items should be included in the list length and are visually styled as disabled at the specified indices.
- Because disabled items are in the list length and a visually perceivable, we need to replicate this behavior for screen reader users.
- aria-disabled="true [is preferred](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-disabled) over the disabled attribute as it includes the item in the count, allows for focus, and announces "disabled" or "dimmed" depending on the browser and screen reader.
